### PR TITLE
Publish localization transform

### PR DESCRIPTION
**Pull request by lorenwel**
These changes publish the latest transform once first localization has occured. 
It's used by my change detection pipeline to obtain the relocalization transform and trigger the start of change detection.

### CODE REVIEW

File: segmatch/src/segmatch.cpp [hunk 1]
```diff
@@ -567,6 +567,10 @@ Time SegMatch::findTimeOfClosestSegmentationPose(const Segment& segment) const {
   return pose_times.at(nearest_neighbour_indices.at(0));
 }
 
+void SegMatch::getLastTransform(Eigen::Matrix4d* out) const {
+  *out = last_transformation_.cast<double>();
```
Comment (rdube) [inline segmatch/src/segmatch.cpp]: add `CHECK_NOTNULL(out);` above

File: segmatch_ros/src/segmatch_worker.cpp [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #include "segmatch_ros/segmatch_worker.hpp"
 
+#include <eigen_conversions/eigen_msg.h>
+
```
Comment (rdube) [inline segmatch_ros/src/segmatch_worker.cpp]: rm vertical space

File: segmatch_ros/package.xml [hunk 3]
```diff
@@ -17,6 +17,7 @@
   <depend>cmake_modules</depend>
   <depend>doxygen_catkin</depend>
   <depend>eigen_catkin</depend>
+  <depend>eigen_conversions</depend>
```
Comment (rdube) [inline segmatch_ros/package.xml]: For which functionality do you exactly need this one? Would be nice to do without (the list of dependencies is already huge! ;)

**Review by rdube — COMMENTED** [review]
@lorenwel that looks good! I would like to check whether we can make it without the new dependency. However that is for both of us not first priority so I can have a look at modifying this later.

File: segmatch_ros/package.xml (hunk 3 above)
Comment (lorenwel) [inline segmatch_ros/package.xml]: This dependency is for conversion from eigen transform to geometry_msgs::Transform.
We could do it ourselves if you really don/t want to have that dependency

**Comment by lorenwel** [comment]
Outdated

**Pull request closed without merging**